src/nvim/generators/gen_options.lua

```diff
@@ -167,8 +167,11 @@ local function dump_option(i, o)
 
   -- An option cannot be both hidden and immutable.
   assert(not o.hidden or not o.immutable)
+  -- Hidden options cannot have a varname.
+  assert(not o.hidden or not o.varname)
+  -- `enable_if` cannot be used with `hidden`.
+  assert(not o.enable_if or not o.hidden)
```

Comment:
the comments are helpful, but might as well pass them as messages too
```suggestion
  assert(not o.hidden or not o.varname, 'hidden options cannot have a varname')
  assert(not o.enable_if or not o.hidden, '`enable_if` cannot be used with `hidden`')
```